packages/components/src/icon.d.ts

```diff
@@ -1,4 +1,3 @@
-// eslint-disable-next-line @typescript-eslint/no-restricted-imports
```

Comment:
I updated eslint config to no longer forbid `FC`

Reply:
I wanted to ask about this. What's the reasoning for this change?

I'm not really a fan of `FC`, because it includes `propTypes`, `deprecatedLegacyContext` and `contextTypes` for backwards compatibility.

When I did some type-level manipulation with components a while ago these properties got in the way a couple of times.

<img width="930" alt="image" src="https://github.com/user-attachments/assets/6b9767e9-143f-4109-9a74-3f127d7dde21">

Reply:
> I'm not really a fan of FC, because it includes propTypes, deprecatedLegacyContext and contextTypes for backwards compatibility.

because:
1. you don't need to declare a return type
1. it's safer, when you return `undefined` from a component.
   > <s>Since React will throw an error, `FC` will notify you about it</s>. Just tested and React no longer throw an error when `undefined` is returned 👀
1. you can also return an array of elements without declaring a return type
   > I guess I did previously mistake when I declared `ReactElement` or `null` as return type. It should be just `(): ReactNode`

Reply:
anyway I changed my mind and prefer a `type` from React

Reply:
fair dos